daemon/container_linux.go

```diff
@@ -803,6 +803,11 @@ func (container *Container) AllocateNetwork() error {
 		return fmt.Errorf("conflicting options: publishing a service and network mode")
 	}
 
+	if mode.IsNetwork() {
+		parts := strings.SplitN(string(mode), ":", 2)
+		networkName = parts[1]
```

Comment:
Rajat,

I don't think so - this invocation path will never create a network - it
just references an existing one.  If you refer to a network that does not
exist, this will fail (which is what we want, right?)

On Tue, Jun 23, 2015 at 7:49 PM, Rajat Chopra <notifications@github.com>
wrote:

> In daemon/container_linux.go
> <https://github.com/docker/docker/pull/14143#discussion_r33113600>:
>
> > @@ -803,6 +803,11 @@ func (container *Container) AllocateNetwork() error {
> >  		return fmt.Errorf("conflicting options: publishing a service and network mode")
> >  	}
> >
> > +	if mode.IsNetwork() {
> > +		parts := strings.SplitN(string(mode), ":", 2)
> > +		networkName = parts[1]
>
> Doesn't the networkDriver need be recalculated or am I missing something.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/14143/files#r33113600>.
>